### ISSUE TITLE
`mandatory` filter not working

### ISSUE THREAD

**Issue by yarsanukayev** [ansible/ansible#10136]
**Issue Type:**
Bug Report

**Ansible Version:**
1.8.2
1.9

**Environment:**
Tested on RHEL5 and on RHEL6.

**Summary:**
`mandatory` filter doesn't seem to work.

**Steps To Reproduce:**
In `ansible.cfg`, set parameter `error_on_undefined_vars` value to `False`.

A playbook:

```
- hosts: meaow
  remote_user: spongebob
  gather_facts: false
  vars:
    - foo: 42
  tasks:
    - copy: "src=/tmp/dummy dest=/tmp/dummy_{{bar | mandatory}}"
```

**Expected Results:**
I expected the following output and no file in destination path.

```
$ ansible-playbook test-mandatory.yml -i hosts

PLAY [meaow] ******************************************************************

TASK: [copy src=/tmp/dummy dest=/tmp/dummy_{{bar | mandatory}}] ***************
fatal: [172.16.0.69] => Mandatory variable not defined.

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/home/squidward/test-mandatory.retry

172.16.0.69                : ok=0    changed=0    unreachable=1    failed=0
```

(I actually wiped out the code from `mandatory` function in `filter_plugins/core.py` leaving only this line so as to force an error and cause failure:

``` python
def mandatory(a):
    raise errors.AnsibleFilterError('Mandatory variable not defined.')
```

)
**Actual Results:**
I'm getting this output and a file `/tmp/dummy_{{bar | mandatory}}` on destination host.

```
PLAY [meaow] ******************************************************************

TASK: [copy src=/tmp/dummy dest=/tmp/dummy_{{bar | mandatory}}] ***************
changed: [172.16.0.69]

PLAY RECAP ********************************************************************
172.16.0.69                : ok=1    changed=1    unreachable=0    failed=0
```

**Comment by bcoca** [comment]
confirmed the bug, the exception is being caught and ignored somewhere up the stack

**Comment by jimi-c** [comment]
I have serious questions as to whether this filter ever worked... Inside that method, `a` is always defined, so using the try/except method for this would never work and an undefined variable passed in as an argument to that would fail before the method was ever reached.

**Comment by bcoca** [comment]
fixed via fe06577ac284de31718c4723809b1d5e1cf5ca4f
